### PR TITLE
Fix ScriptTextEditor encapsulation

### PR DESCRIPTION
**Pull request by Chaosus**
which I sadly broke in #34402

### CODE REVIEW

File: editor/plugins/script_text_editor.cpp [hunk 1]
```diff
@@ -966,6 +966,10 @@ void ScriptTextEditor::_lookup_symbol(const String &p_symbol, int p_row, int p_c
 	}
 }
 
+CodeTextEditor* ScriptTextEditor::get_code_editor() {
```
Comment (YeldhamDev) [inline editor/plugins/script_text_editor.cpp]: The asterisk should be in the right.

File: editor/plugins/script_text_editor.h [hunk 2]
```diff
@@ -186,8 +187,7 @@ class ScriptTextEditor : public ScriptEditorBase {
 	void drop_data_fw(const Point2 &p_point, const Variant &p_data, Control *p_from);
 
 public:
-	CodeTextEditor *code_editor;
-
+	CodeTextEditor* get_code_editor();
```
Comment (YeldhamDev) [inline editor/plugins/script_text_editor.h]: Same here.

File: editor/plugins/script_text_editor.cpp (hunk 1 above)
Comment (Chaosus) [reply editor/plugins/script_text_editor.cpp]: I'm fixed my clang-format...

**Review by Paulb23 — COMMENTED** [review]
This still breaks it, as `CodeTextEditor` is being used in `script_editor_plugin`. 

The method `update_toggle_scripts_button` needs to exist on `ScriptTextEditor` not `CodeTextEditor`.

**Comment by Chaosus** [comment]
@Paulb23 Alright, I rewrote that part. Can you check again?

File: editor/plugins/script_editor_plugin.cpp [hunk 3]
```diff
@@ -1141,14 +1138,13 @@ void ScriptEditor::_menu_option(int p_option) {
 		} break;
 		case TOGGLE_SCRIPTS_PANEL: {
 			if (current) {
-				CodeTextEditor *code_editor = NULL;
 				ScriptTextEditor *editor = dynamic_cast<ScriptTextEditor *>(current);
```
Comment (Paulb23) [inline editor/plugins/script_editor_plugin.cpp]: Does this need to handle `TextEditor` and `VisualScripts`? 

To keep constancy should use ` Object::cast_to<ScriptTextEditor>(current)`.

**Review by Paulb23 — COMMENTED** [review]
Thanks for updating, code looks good to me.  One minor inline note.

File: editor/plugins/script_editor_plugin.cpp (hunk 3 above)
Comment (Chaosus) [reply editor/plugins/script_editor_plugin.cpp]: > Does this need to handle TextEditor and VisualScripts?

Exactly.

> To keep constancy should use Object::cast_to<ScriptTextEditor>(current).

Fixed.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**